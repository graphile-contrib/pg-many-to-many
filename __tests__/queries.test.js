/* eslint-disable no-console */
const fs = require("fs");
const util = require("util");
const path = require("path");
const { graphql } = require("graphql");
const { withPgClient } = require("./helpers");
const { createPostGraphileSchema } = require("postgraphile-core");
const { printSchema } = require("graphql/utilities");
const debug = require("debug")("graphile-build:schema");

const readFile = util.promisify(fs.readFile);

const getSqlSchemas = () => fs.readdirSync(path.resolve(__dirname, "schemas"));
const getFixturesForSqlSchema = sqlSchema =>
  fs.existsSync(
    path.resolve(__dirname, "schemas", sqlSchema, "fixtures", "queries")
  )
    ? fs.readdirSync(
        path.resolve(__dirname, "schemas", sqlSchema, "fixtures", "queries")
      )
    : [];
const readFixtureForSqlSchema = (sqlSchema, fixture) =>
  readFile(
    path.resolve(
      __dirname,
      "schemas",
      sqlSchema,
      "fixtures",
      "queries",
      fixture
    ),
    "utf8"
  );

const queryResult = async (sqlSchema, fixture) => {
  return await withPgClient(async pgClient => {
    const data = await readFile(
      path.resolve(__dirname, "schemas", sqlSchema, "data.sql"),
      "utf8"
    );
    await pgClient.query(data);
    const PgManyToManyPlugin = require("../index");
    const gqlSchema = await createPostGraphileSchema(pgClient, [sqlSchema], {
      appendPlugins: [PgManyToManyPlugin],
    });
    debug(`${sqlSchema}: ${printSchema(gqlSchema)}`);
    const query = await readFixtureForSqlSchema(sqlSchema, fixture);
    return await graphql(gqlSchema, query, null, {
      pgClient: pgClient,
    });
  });
};

const sqlSchemas = getSqlSchemas();
describe.each(sqlSchemas)("schema=%s", sqlSchema => {
  const fixtures = getFixturesForSqlSchema(sqlSchema);
  if (fixtures.length > 0) {
    test.each(fixtures)("query=%s", async fixture => {
      const result = await queryResult(sqlSchema, fixture);
      if (result.errors) {
        console.log(result.errors.map(e => e.originalError));
      }
      expect(result).toMatchSnapshot();
    });
  }
});
