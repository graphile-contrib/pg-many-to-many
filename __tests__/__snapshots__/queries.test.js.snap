// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema=a query=edges.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
                "node": Object {
                  "barId": 12,
                  "barName": "Bar 12",
                  "nodeId": "WyJiYXJzIiwxMl0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=a query=nodes.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
              Object {
                "barId": 12,
                "barName": "Bar 12",
                "nodeId": "WyJiYXJzIiwxMl0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=b query=edges.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "createdAt": "2018-01-01T08:00:00-04:00",
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
              Object {
                "createdAt": "2018-01-02T08:00:00-04:00",
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
                "node": Object {
                  "barId": 12,
                  "barName": "Bar 12",
                  "nodeId": "WyJiYXJzIiwxMl0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "createdAt": "2018-01-03T08:00:00-04:00",
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=b query=nodes.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
              Object {
                "barId": 12,
                "barName": "Bar 12",
                "nodeId": "WyJiYXJzIiwxMl0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=c query=edges.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                      "node": Object {
                        "jBarId": 11,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDFd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                  },
                  "totalCount": 1,
                },
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
                      "node": Object {
                        "jBarId": 12,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDJd",
                      },
                    },
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
                      "node": Object {
                        "jBarId": 12,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDNd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
                  },
                  "totalCount": 2,
                },
                "node": Object {
                  "barId": 12,
                  "barName": "Bar 12",
                  "nodeId": "WyJiYXJzIiwxMl0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                      "node": Object {
                        "jBarId": 11,
                        "jFooId": 2,
                        "nodeId": "WyJqdW5jdGlvbnMiLDRd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                  },
                  "totalCount": 1,
                },
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=c query=nodes.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
              Object {
                "barId": 12,
                "barName": "Bar 12",
                "nodeId": "WyJiYXJzIiwxMl0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=d query=edges.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                      "node": Object {
                        "createdAt": "2018-01-01T08:00:00-04:00",
                        "jBarId": 11,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDFd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxXV0=",
                  },
                  "totalCount": 1,
                },
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
                      "node": Object {
                        "createdAt": "2018-01-02T08:00:00-04:00",
                        "jBarId": 12,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDJd",
                      },
                    },
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
                      "node": Object {
                        "createdAt": "2018-01-04T08:00:00-04:00",
                        "jBarId": 12,
                        "jFooId": 1,
                        "nodeId": "WyJqdW5jdGlvbnMiLDNd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFszXV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsyXV0=",
                  },
                  "totalCount": 2,
                },
                "node": Object {
                  "barId": 12,
                  "barName": "Bar 12",
                  "nodeId": "WyJiYXJzIiwxMl0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [
              Object {
                "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
                "junctionsByJBarId": Object {
                  "edges": Array [
                    Object {
                      "cursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                      "node": Object {
                        "createdAt": "2018-01-03T08:00:00-04:00",
                        "jBarId": 11,
                        "jFooId": 2,
                        "nodeId": "WyJqdW5jdGlvbnMiLDRd",
                      },
                    },
                  ],
                  "pageInfo": Object {
                    "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                    "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFs0XV0=",
                  },
                  "totalCount": 1,
                },
                "node": Object {
                  "barId": 11,
                  "barName": "Bar 11",
                  "nodeId": "WyJiYXJzIiwxMV0=",
                },
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "edges": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=d query=nodes.graphql 1`] = `
Object {
  "data": Object {
    "allFoos": Object {
      "nodes": Array [
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
              Object {
                "barId": 12,
                "barName": "Bar 12",
                "nodeId": "WyJiYXJzIiwxMl0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMl1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 2,
          },
          "fooId": 1,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [
              Object {
                "barId": 11,
                "barName": "Bar 11",
                "nodeId": "WyJiYXJzIiwxMV0=",
              },
            ],
            "pageInfo": Object {
              "endCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
              "startCursor": "WyJwcmltYXJ5X2tleV9hc2MiLFsxMV1d",
            },
            "totalCount": 1,
          },
          "fooId": 2,
        },
        Object {
          "barsByJunctionJFooIdAndJBarId": Object {
            "nodes": Array [],
            "pageInfo": Object {
              "endCursor": null,
              "startCursor": null,
            },
            "totalCount": 0,
          },
          "fooId": 3,
        },
      ],
    },
  },
}
`;

exports[`schema=e query=multiple-tags-fields.graphql 1`] = `
Object {
  "data": Object {
    "allPeople": Object {
      "nodes": Array [
        Object {
          "personName": "Alice",
          "tags": Object {
            "nodes": Array [
              Object {
                "tagName": "awesome",
              },
            ],
          },
          "teams": Object {
            "nodes": Array [
              Object {
                "tags": Object {
                  "nodes": Array [
                    Object {
                      "tagName": "strange",
                    },
                  ],
                },
                "teamName": "Development",
              },
            ],
          },
        },
        Object {
          "personName": "Bob",
          "tags": Object {
            "nodes": Array [
              Object {
                "tagName": "strange",
              },
            ],
          },
          "teams": Object {
            "nodes": Array [
              Object {
                "tags": Object {
                  "nodes": Array [
                    Object {
                      "tagName": "high-performing",
                    },
                  ],
                },
                "teamName": "Sales",
              },
            ],
          },
        },
        Object {
          "personName": "Carol",
          "tags": Object {
            "nodes": Array [
              Object {
                "tagName": "high-performing",
              },
            ],
          },
          "teams": Object {
            "nodes": Array [
              Object {
                "tags": Object {
                  "nodes": Array [
                    Object {
                      "tagName": "awesome",
                    },
                  ],
                },
                "teamName": "Marketing",
              },
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`schema=f query=f.graphql 1`] = `
Object {
  "data": Object {
    "allPeople": Object {
      "nodes": Array [
        Object {
          "followers": Object {
            "nodes": Array [
              Object {
                "name": "Bob",
              },
              Object {
                "name": "Eve",
              },
            ],
          },
          "following": Object {
            "nodes": Array [],
          },
          "name": "Alice",
        },
        Object {
          "followers": Object {
            "nodes": Array [
              Object {
                "name": "Eve",
              },
            ],
          },
          "following": Object {
            "nodes": Array [
              Object {
                "name": "Alice",
              },
            ],
          },
          "name": "Bob",
        },
        Object {
          "followers": Object {
            "nodes": Array [],
          },
          "following": Object {
            "nodes": Array [
              Object {
                "name": "Alice",
              },
              Object {
                "name": "Bob",
              },
            ],
          },
          "name": "Eve",
        },
      ],
    },
  },
}
`;

exports[`schema=p query=edge-fields.graphql 1`] = `
Object {
  "data": Object {
    "allTeams": Object {
      "nodes": Array [
        Object {
          "members": Object {
            "edges": Array [
              Object {
                "createdAt": "2018-01-01T08:00:00-04:00",
                "node": Object {
                  "personName": "Person1",
                },
              },
              Object {
                "createdAt": "2018-01-03T08:00:00-04:00",
                "node": Object {
                  "personName": "Person2",
                },
              },
            ],
          },
          "teamName": "Team1",
        },
        Object {
          "members": Object {
            "edges": Array [
              Object {
                "createdAt": "2018-01-02T08:00:00-04:00",
                "node": Object {
                  "personName": "Person1",
                },
              },
            ],
          },
          "teamName": "Team2",
        },
        Object {
          "members": Object {
            "edges": Array [],
          },
          "teamName": "Team3",
        },
      ],
    },
  },
}
`;

exports[`schema=p query=many-to-many.graphql 1`] = `
Object {
  "data": Object {
    "a": Object {
      "nodes": Array [
        Object {
          "personName": "Person1",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [
              Object {
                "teamName": "Team1",
              },
              Object {
                "teamName": "Team2",
              },
            ],
          },
        },
        Object {
          "personName": "Person2",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [
              Object {
                "teamName": "Team1",
              },
            ],
          },
        },
        Object {
          "personName": "Person3",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [],
          },
        },
      ],
    },
    "b": Object {
      "nodes": Array [
        Object {
          "members": Object {
            "nodes": Array [
              Object {
                "personName": "Person1",
              },
              Object {
                "personName": "Person2",
              },
            ],
          },
          "teamName": "Team1",
        },
        Object {
          "members": Object {
            "nodes": Array [
              Object {
                "personName": "Person1",
              },
            ],
          },
          "teamName": "Team2",
        },
        Object {
          "members": Object {
            "nodes": Array [],
          },
          "teamName": "Team3",
        },
      ],
    },
  },
}
`;

exports[`schema=t query=edge-fields.graphql 1`] = `
Object {
  "data": Object {
    "allTeams": Object {
      "nodes": Array [
        Object {
          "members": Object {
            "edges": Array [
              Object {
                "membershipsByPersonId": Object {
                  "nodes": Array [
                    Object {
                      "endAt": null,
                      "startAt": "2018-01-01T08:00:00-04:00",
                    },
                  ],
                },
                "node": Object {
                  "personName": "Person1",
                },
              },
              Object {
                "membershipsByPersonId": Object {
                  "nodes": Array [
                    Object {
                      "endAt": null,
                      "startAt": "2018-01-03T08:00:00-04:00",
                    },
                  ],
                },
                "node": Object {
                  "personName": "Person2",
                },
              },
            ],
          },
          "teamName": "Team1",
        },
        Object {
          "members": Object {
            "edges": Array [
              Object {
                "membershipsByPersonId": Object {
                  "nodes": Array [
                    Object {
                      "endAt": "2018-01-03T08:00:00-04:00",
                      "startAt": "2018-01-02T08:00:00-04:00",
                    },
                    Object {
                      "endAt": "2018-01-05T08:00:00-04:00",
                      "startAt": "2018-01-04T08:00:00-04:00",
                    },
                  ],
                },
                "node": Object {
                  "personName": "Person1",
                },
              },
            ],
          },
          "teamName": "Team2",
        },
        Object {
          "members": Object {
            "edges": Array [],
          },
          "teamName": "Team3",
        },
      ],
    },
  },
}
`;

exports[`schema=t query=many-to-many.graphql 1`] = `
Object {
  "data": Object {
    "a": Object {
      "nodes": Array [
        Object {
          "personName": "Person1",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [
              Object {
                "teamName": "Team1",
              },
              Object {
                "teamName": "Team2",
              },
            ],
          },
        },
        Object {
          "personName": "Person2",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [
              Object {
                "teamName": "Team1",
              },
            ],
          },
        },
        Object {
          "personName": "Person3",
          "teamsByMembershipPersonIdAndTeamId": Object {
            "nodes": Array [],
          },
        },
      ],
    },
    "b": Object {
      "nodes": Array [
        Object {
          "members": Object {
            "nodes": Array [
              Object {
                "personName": "Person1",
              },
              Object {
                "personName": "Person2",
              },
            ],
          },
          "teamName": "Team1",
        },
        Object {
          "members": Object {
            "nodes": Array [
              Object {
                "personName": "Person1",
              },
            ],
          },
          "teamName": "Team2",
        },
        Object {
          "members": Object {
            "nodes": Array [],
          },
          "teamName": "Team3",
        },
      ],
    },
  },
}
`;
